libraries/chain/host_context.cpp

```diff
@@ -39,54 +39,83 @@ host_context::~host_context() = default;
-   EOS_ASSERT(code != nullptr, sync_call_validate_exception,
-              "sync call's receiver account ${r} does not exist", ("r", call_receiver));
-
-   EOS_ASSERT(flags <= static_cast<uint64_t>(sync_call_flags::all_allowed_bits), sync_call_validate_exception,  // all but `std::bit_width(all_allowed_bits)` LSBs must be 0s
-              "only ${bits} least significant bits of sync call's flags (${flags}) can be set", ("bits", std::bit_width(static_cast<uint64_t>(sync_call_flags::all_allowed_bits)))("flags", flags));
+   uint32_t ordinal = trace.call_traces.size();
+   get_call_trace(ordinal).call_ordinal = ordinal;
 
    auto handle_exception = [&](const auto& e)
    {
-      // TBD need to store the exception in the sync call trace
+      auto& call_trace = get_call_trace(ordinal);
+      call_trace.error_code = controller::convert_exception_to_error_code(e);
+      call_trace.except = e;
+      finalize_call_trace(call_trace, start);
       throw;
    };
 
+   auto handle_call_failure = [&]()
+   {
+      auto& call_trace = get_call_trace(ordinal);
+      call_trace.error_id = -1;
+      finalize_call_trace(call_trace, start);
+      trx_context.checktime();
+      return -1;
+   };
+
    last_sync_call_return_value.clear(); // reset for current sync call
    uint32_t return_value_size  = 0;
 
    try {
       try {
+         const uint32_t depth = sync_call_depth + 1;
+         const auto max_depth = control.get_global_properties().configuration.max_sync_call_depth;
+         EOS_ASSERT(depth <= max_depth, sync_call_depth_exception,
+                    "reached sync call max call depth ${max_depth}", ("max_depth", max_depth));
+
+         const auto max_data_size = control.get_global_properties().configuration.max_sync_call_data_size;
+         EOS_ASSERT(data.size() <= max_data_size, sync_call_call_data_exception,
+                    "sync call call data size must be less or equal to ${max_data_size} bytes", ("max_data_size", max_data_size));
+
+         const auto* code = control.db().find<account_object, by_name>(call_receiver);
+         EOS_ASSERT(code != nullptr, sync_call_validate_exception,
+                    "sync call's receiver account ${r} does not exist", ("r", call_receiver));
+
+         EOS_ASSERT(flags <= static_cast<uint64_t>(sync_call_flags::all_allowed_bits), sync_call_validate_exception,  // all but `std::bit_width(all_allowed_bits)` LSBs must be 0s
+                    "only ${bits} least significant bits of sync call's flags (${flags}) can be set", ("bits", std::bit_width(static_cast<uint64_t>(sync_call_flags::all_allowed_bits)))("flags", flags));
+
          const account_metadata_object* receiver_account = &db.get<account_metadata_object, by_name>( call_receiver);
          if (receiver_account->code_hash.empty()) {
-            // TBD store the info in sync call trace
-            ilog("receiver_account->code_hash empty");
-            return return_value_size;
+            return handle_call_failure();
```

Comment:
In this case the console marker will not be created. Seems like creation of console marker should be moved way up earlier in the function

Reply:
I put it here thinking if we could not make a sync call for any reasons before creating the sync call context, we should not put the sync call position in the markers, since no call was made.

Reply:
No call was made but you added a trace for the call still. That means it becomes impossible to use the markers properly because the number of markers will not equal the number of sync call traces

Reply:
Yes, you are right. I will put them together.

Reply:
https://github.com/AntelopeIO/spring/pull/1333/commits/b361003952ccbcdd944c4e49bd31d7067542ba97